File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C++/s573993837_comments_removed.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <vector>
#include <cstdint>
using namespace std;

static int L, R, top, bottom;

class InputReader {
private:
    istream& in;
    string line;
    istringstream currentStream;
public:
    InputReader(istream& _in) : in(_in) {}
    string next() {
        while (true) {
            if (currentStream.eof()) {
                if (!getline(in, line)) {
                    throw runtime_error("Error reading input");
                }
                currentStream.str(line);
                currentStream.clear();
            }
            string token;
            if (currentStream >> token) {
                return token;
            }
        }
    }
    int nextInt() { return stoi(next()); }
    long long nextLong() { return stoll(next()); }
    double nextDouble() { return stod(next()); }
};

void swap(char arr[], int i, int j) {
    char temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

int num(int n) {
    int a = 0;
    while (n > 0) {
        a += (n & 1);
        n >>= 1;
    }
    return a;
}

struct Pair {
    int u, v, i, val;
    Pair(int a, int b, int i) : u(a), v(b), i(i) {}
};

bool isPrime(int n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;
    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) return false;
    }
    return true;
}

long long gcd(long long a, long long b) {
    return b == 0 ? a : gcd(b, a % b);
}

long long fast_pow(long long base, long long n, long long M) {
    if (n == 0) return 1;
    long long halfn = fast_pow(base, n / 2, M);
    return (n % 2 == 0) ? 
        (halfn * halfn) % M : 
        ((halfn * halfn) % M * base) % M;
}

long long modInverse(long long n, long long M) {
    return fast_pow(n, M - 2, M);
}

void solve(InputReader& in, ostream& out) {
    int t = 1;
    while (t-- > 0) {
        long long n = in.nextLong(), m = in.nextLong();
        if (n == 0) {
            out << "0\n";
            continue;
        }
        if (n == m) {
            out << "0\n";
            continue;
        }
        if (n > m) {
            n %= m;
        }
        out << min(m - n, n) << '\n';
    }
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    InputReader in(cin);
    solve(in, cout);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s573993837_comments_removed.py
class InputReader:
    def __init__(self, stream):
        self.reader = stream
        self.tokens = []
    
    def next(self):
        while not self.tokens:
            line = self.reader.readline()
            if not line:
                return None
            self.tokens = line.split()
        return self.tokens.pop(0)
    
    def nextInt(self):
        return int(self.next())
    
    def nextLong(self):
        return int(self.next())
    
    def nextDouble(self):
        return float(self.next())

def swap(chrr, i, j):
    temp = chrr[i]
    chrr[i] = chrr[j]
    chrr[j] = temp

def num(n):
    a = 0
    while n > 0:
        a += (n & 1)
        n >>= 1
    return a

class Pair:
    def __init__(self, a, b, i):
        self.u = a
        self.v = b
        self.i = i
        self.val = 0

def isPrime(n):
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b) if a > b else gcd(a, b % a)

def fast_pow(base, n, M):
    if n == 0:
        return 1
    if n == 1:
        return base
    halfn = fast_pow(base, n // 2, M)
    if n % 2 == 0:
        return (halfn * halfn) % M
    else:
        return (( (halfn * halfn) % M ) * base ) % M

def modInverse(n, M):
    return fast_pow(n, M - 2, M)

def feedArrLong(arr, sc):
    for i in range(len(arr)):
        arr[i] = sc.nextLong()

def feedArrDouble(arr, sc):
    for i in range(len(arr)):
        arr[i] = sc.nextDouble()

def feedArrInt(arr, sc):
    for i in range(len(arr)):
        arr[i] = sc.nextInt()

def feedArrString(arr, sc):
    for i in range(len(arr)):
        arr[i] = sc.next()

def printArr(arr):
    return ' '.join(map(str, arr)) + ' '

def solve(sc, pw):
    t = 1
    while t > 0:
        t -= 1
        n = sc.nextLong()
        m = sc.nextLong()
        if n == 0:
            print(0, file=pw)
            continue
        if n == m:
            print(0, file=pw)
            continue
        if n > m:
            n %= m
        res = min(m - n, n)
        print(res, file=pw)

if __name__ == '__main__':
    import sys
    in_reader = InputReader(sys.stdin)
    out = sys.stdout
    solve(in_reader, out)
    # End of Code
